weblab/static/js/upload.js

```diff
@@ -88,8 +88,8 @@ Upload.prototype = {
       error = "there is already a file with the name '" + name + "' - please remove that first.", "error";
     } else if (this.reserved_names.indexOf(name) != -1) {
       error = "the name '" + name + "' is reserved for system use; please choose another file name.", "error";
-    } else if (!/^[a-zA-Z0-9._]+$/.test(name)) {
-      error = "the name '" + name + "' contains reserved characters; only alpha-numeric characters, underscores and periods are allowed.", "error";
+    } else if (!/^[\w._: \-]+$/.test(name)) {
```

Comment:
Is there nothing in Django or the standard library (e.g. urllib) we can use for this check instead of building our own regex?

Reply:
Looking into this a bit I find additional checks as well, e.g. at most 244 characters for windows compatibility

Reply:
It's a bit ugly because these things are being checked both by the JS code and Django. At least now those regexs are more similar to each other!